lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -4279,6 +4288,11 @@ LogicalResult FIRRTLLowering::visitExpr(XMRDerefOp op) {
   return setLoweringTo<seq::ToClockOp>(op, readXmr);
 }
 
+// Do nothing when lowering this operation.  We need to handle this at its usage
+// sites to guarantee creation of `sv::TimeOp` that is colocated with the
+// `sv::FWriteOp`.
+LogicalResult FIRRTLLowering::visitExpr(TimeOp op) { return success(); }
+
```

Comment:
I'm wondering if this is strictly necessary :thinking:. Since `$time` should be essentially an integer value in Verilog, spilling might just work. I'm also totally fine with treating this as a special case here.

Reply:
What I was worried about was if this somehow got CSE'd very far away from the `printf`. As long as it is in the same `always` block as the `printf`, it should be completely fine, even if spilled. I was worried about it getting outside that `always` block to produce something like:

```
wire _GEN = $time;

always @posedge(clock)
  $dipslay("%t", _GEN);

always @posedge(clock)
  $dipslay("%t", _GEN);
```

Same issue for the more complicate multi-clock case, or similar.